### PR TITLE
Modulo 4

### PR DESCRIPTION
**Pull request by hhualiang**
Project reformat following Clean Architecture guidelines
Lint package and rules added

### CODE REVIEW

**Review by LeonardoMolleker — APPROVED** [review]

File: lib/src/data/model/movie_genre_model.dart
Comment (matiasilveiro-gl) [inline lib/src/data/model/movie_genre_model.dart:1]: I'd move this class to the `domain/entity` layer

File: lib/src/data/model/movie_model.dart
Comment (matiasilveiro-gl) [inline lib/src/data/model/movie_model.dart:1]: I'd move this class to the `domain/entity` layer

**Review by matiasilveiro-gl — COMMENTED** [review]
The overall architecture looks good!

File: lib/src/presentation/widget/home_screen/home_screen_movie_row.dart [hunk 1]
```diff
@@ -21,7 +21,7 @@ class HomeScreenMovieRow extends StatelessWidget {
         onTap: () {
           Navigator.push(
             context,
-            MaterialPageRoute(
+            MaterialPageRoute<dynamic>(
```
Comment (matiasilveiro-gl) [inline lib/src/presentation/widget/home_screen/home_screen_movie_row.dart:24]: Avoid `dynamic`!

**Review by matiasilveiro-gl — APPROVED** [review]
LGTM

**Review by Kevin-Guillen — APPROVED** [review]

**Pull request merged**